lib/traject_plus/macros.rb

```diff
@@ -15,25 +15,60 @@ def transform_values(context, hash)
     # try a bunch of macros and short-circuit after one returns values
     def first(*macros)
       lambda do |record, accumulator, context|
-        macros.lazy.map do |lambda|
-          lambda.call(record, accumulator, context)
+        macros.lazy.map do |block|
+          block.call(record, accumulator, context)
         end.reject(&:blank?).first
       end
     end
 
+    def accumulate(&block)
+      lambda do |record, accumulator, context|
+        Array(block.call(record, context)).each do |v|
+          accumulator << v if v.present?
+        end
+      end
+    end
+
     # only accumulate values if a condition is met
-    def conditional(condition, lambda)
+    def conditional(condition, block)
       lambda do |record, accumulator, context|
         if condition.call(record, context)
-          lambda.call(record, accumulator, context)
+          block.call(record, accumulator, context)
         end
       end
     end
 
     def from_settings(field)
-      lambda { |_record, accumulator, context|
-        accumulator << context.settings.fetch(field)
-      }
+      accumulate do |record, context|
+        context.settings.fetch(field)
+      end
+    end
+
+    def copy(field)
+      accumulate do |_record, context|
+        Array(context.output_hash[field])
+      end
+    end
+
+    # apply the same mapping to multiple fields
+    def to_fields(fields, mapping_method)
+      fields.each { |field| to_field field, mapping_method }
+    end
+
+    def to_field(field_name, aLambda = nil, extract: nil, transform: nil, **namedArgs, &block)
+      @index_steps << TrajectPlus::Indexer::ToFieldStep.new(field_name, extract || aLambda, transform || block, Traject::Util.extract_caller_location(caller.first), **namedArgs)
+    end
+
+    def compose(fieldname = nil, aLambda = nil, extract: nil, transform: nil, &block)
```

Comment:
I'm getting an error trying to use compose to assign anything other than a straight up passed string or a Traject core method / lambda to this. I.e. the following in a mapping config file leveraging these additions:

```ruby
compose 'test', ->(record, accumulator, _context) { accumulator << record } do
  to_field 'composed', extract_xml("/TEI/teiHeader/encodingDesc/projectDesc/p")
end
```

Generates this error:

```Error loading configuration file manifestConfig.rb:30 NoMethodError:undefined method `extract_xml' for #<Traject::Indexer:0x007fdd9c17ef40>```.

It seems like `compose` functions correctly, I'm just uncertain where to add references to the TrajectPlus libraries if I want to use this method for mapping the outputs of those to fields in the composed field.